command/aarelayer/service/aa_state.go

```diff
@@ -13,58 +15,163 @@ type AATxState interface {
+	// Get all queued transactions
+	GetAllQueued() ([]*AAStateTransaction, error)
+	// Update modifies the metadata for the AA transaction
+	Update(stateTx *AAStateTransaction) error
 }
 
+var (
+	pendingBucket  = []byte("pending")
+	queuedBucket   = []byte("queued")
+	finishedBucket = []byte("finished")
+)
+
 var _ AATxState = (*aaTxState)(nil)
 
 type aaTxState struct {
-	// TODO: will be replaced with boldDB in next PR/task
-	mutex sync.RWMutex
-	items map[string]*AAStateTransaction
+	db *bolt.DB
 }
 
-func NewAATxState() (*aaTxState, error) {
-	return &aaTxState{
-		items: make(map[string]*AAStateTransaction),
-	}, nil
+func NewAATxState(dbFilePath string) (*aaTxState, error) {
+	state := &aaTxState{}
+
+	if err := state.init(dbFilePath); err != nil {
+		return nil, err
+	}
+
+	return state, nil
 }
 
 func (s *aaTxState) Add(tx *AATransaction) (*AAStateTransaction, error) {
-	s.mutex.Lock()
-	s.mutex.Unlock()
-
-	// TODO: bolDB implementation will be in next PR/task
-	id := uuid.NewString()
 	ntx := &AAStateTransaction{
-		ID:     id,
+		ID:     uuid.NewString(),
 		Tx:     tx,
 		Status: StatusPending,
+		Time:   time.Now().Unix(),
+	}
+
+	value, err := json.Marshal(ntx)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := s.db.Update(func(tx *bolt.Tx) error {
+		return tx.Bucket(pendingBucket).Put([]byte(ntx.ID), value)
+	}); err != nil {
+		return nil, err
 	}
-	s.items[id] = ntx
 
 	return ntx, nil
 }
 
-func (s *aaTxState) Get(id string) (*AAStateTransaction, error) {
-	s.mutex.RLock()
-	s.mutex.RUnlock()
+func (s *aaTxState) Get(id string) (result *AAStateTransaction, err error) {
+	if err := s.db.View(func(tx *bolt.Tx) error {
+		idb := []byte(id)
+
+		for _, bucket := range [][]byte{pendingBucket, queuedBucket, finishedBucket} {
+			value := tx.Bucket(bucket).Get(idb)
+			if value != nil {
+				return json.Unmarshal(value, &result)
+			}
+		}
+
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (s *aaTxState) GetAllPending() ([]*AAStateTransaction, error) {
+	return s.getAllFromBucket(pendingBucket)
+}
 
-	// TODO: bolDB implementation will be in next PR/task
-	return s.items[id], nil
+func (s *aaTxState) GetAllQueued() ([]*AAStateTransaction, error) {
+	return s.getAllFromBucket(queuedBucket)
 }
 
-func (s *aaTxState) Update(id string, fn func(tx *AAStateTransaction)) error {
-	s.mutex.Lock()
-	s.mutex.Unlock()
+func (s *aaTxState) Update(stateTx *AAStateTransaction) error {
+	return s.db.Update(func(tx *bolt.Tx) error {
+		idb := []byte(stateTx.ID)
+		newStatusBacket := map[string][]byte{
+			StatusPending:   pendingBucket,
+			StatusQueued:    queuedBucket,
+			StatusCompleted: finishedBucket,
+			StatusFailed:    finishedBucket,
+		}[stateTx.Status] // this is just fancy switch case
+
+		// check if item exist in another bucket, and if it is, delete item from the old bucket
+		for _, bucket := range [][]byte{pendingBucket, queuedBucket, finishedBucket} {
```

Comment:
Move this list of all buckets to a global variable (similar what we do in `state.go` in `polybft`).

Reply:
fixed https://github.com/0xPolygon/polygon-edge/pull/1261/commits/735ad3cc881372c7d40c5d60d1a4560623233f86